database/migrations/default/1750435376415_ems_person_match_auto/up.sql

```diff
@@ -0,0 +1,437 @@
+            return NEW;
+        --
+        --  If there is one matching crash ID
+        --
+        ELSIF array_length(NEW.matched_crash_pks, 1) = 1 THEN
+            raise debug 'Reset EMS ID % crash_match_status to matched_by_automation', NEW.id;
+            NEW.crash_pk = NEW.matched_crash_pks[1];
+            NEW.crash_match_status = 'matched_by_automation';
+            -- don't return here so that we proceed to person matching
+        ELSE
+            raise debug 'Reset EMS ID % to multiple_matches_by_automation', NEW.id;
+            NEW.crash_pk = null;
+            NEW.crash_match_status = 'multiple_matches_by_automation';
+            NEW.person_id = NULL;
+            NEW.person_match_status = 'unmatched';
+            NEW.matched_person_ids = NULL;
+            NEW._match_event_name = null;
+            return new;
+        END IF;
+    END IF;
+    
+    --
+    -- Begin automated person-level matching
+    --
+    IF NEW._match_event_name in (
+        'match_crash_by_automation',
+        'sync_crash_pk_on_person_match',
+        'reset_crash_match'
+    )
+        AND new.crash_pk IS NOT NULL THEN
+        --
+        -- reset _match_event_name in outermost block to avoid repetition
+        --
+        NEW._match_event_name = null;
+
+        raise debug 'Doing person matching for EMS record ID % matched to crash %', NEW.id, NEW.crash_pk;
+
+        if NEW.person_match_status = 'matched_by_manual_qa' 
+            or NEW.pcr_patient_age is NULL
+            or NEW.pcr_patient_gender is NULL
+            or NEW.pcr_patient_race is NULL
+            THEN
+                -- ignore this update / nothing to do
+                raise debug 'Skipping person matching for EMS ID %', NEW.id;
+                return new;
+        ELSE
+            matching_person_ids = find_matching_person_ids(NEW);
+
+            IF array_length(matching_person_ids, 1) IS NULL THEN
+                -- no match
+                raise debug 'No person match found for EMS ID %', NEW.id;
+                NEW.person_match_status = 'unmatched';
+                NEW.matched_person_ids = NULL;
+                NEW.person_id = NULL;
+                return NEW;
+            ELSIF array_length(matching_person_ids, 1) = 1 THEN
+                raise debug 'One person match found for EMS ID %', NEW.id;
+                NEW.matched_person_ids = matching_person_ids;
+                NEW.person_id = matching_person_ids[1];
+                NEW.person_match_status = 'matched_by_automation';
+                return NEW;
+            ELSE
+                raise debug 'Multiple person matches found for EMS ID %', NEW.id;
+                NEW.matched_person_ids = matching_person_ids;
+                NEW.person_match_status = 'multiple_matches_by_automation';
+                NEW.person_id = NULL;
+                return NEW;
+            END IF;
+        END IF;
+    END IF;
+    -- todo: do we need this line?
+    NEW._match_event_name = null;
```

Comment:
how do we get to this line? because if its not null, and not in the list of event names, isnt an exception thrown?

Reply:
just saw this and +1 - that added context that Chia suggested would be helpful!